`go mod vendor` excludes required C header files

When using `go mod vendor` to package dependencies in sing-box (common in offline build environments), the `include/` is stripped away. This causes:

```
# github.com/sagernet/cronet-go
vendor/github.com/sagernet/cronet-go/buffer_callback_cgo.go:7:11: fatal error: 'cronet_c.h' file not found
    7 |  #include <cronet_c.h>
      |           ^~~~~~~~~~   
```

```
❯ tree vendor/github.com/sagernet/cronet-go/
vendor/github.com/sagernet/cronet-go/
├── all
│   ├── android_386.go
│   ├── android_amd64.go
│   ├── android_arm64.go
│   ├── android_arm.go
│   ├── darwin_amd64.go
│   ├── darwin_arm64.go
│   ├── ios_amd64_simulator.go
│   ├── ios_arm64.go
│   ├── ios_arm64_simulator.go
│   ├── LICENSE
│   ├── linux_386.go
│   ├── linux_386_musl.go
│   ├── linux_amd64.go
│   ├── linux_amd64_musl.go
│   ├── linux_arm64.go
│   ├── linux_arm64_musl.go
│   ├── linux_arm.go
│   ├── linux_arm_musl.go
│   ├── linux_loong64.go
│   ├── linux_loong64_musl.go
│   ├── linux_mips64le.go
│   ├── linux_mipsle.go
│   ├── linux_mipsle_musl.go
│   ├── linux_riscv64.go
│   ├── linux_riscv64_musl.go
│   ├── tvos_amd64_simulator.go
│   ├── tvos_arm64.go
│   ├── tvos_arm64_simulator.go
│   ├── windows_amd64.go
│   └── windows_arm64.go
├── bidirectional_conn.go
├── bidirectional_stream_cgo.go
├── bidirectional_stream_impl_cgo.go
├── bidirectional_stream_impl_purego.go
├── bidirectional_stream_map.go
├── bidirectional_stream_purego.go
├── buffer_callback_cgo.go
├── buffer_callback_impl.go
├── buffer_callback_impl_purego.go
├── buffer_callback_purego.go
├── buffer_cgo.go
├── buffer_impl_cgo.go
├── buffer_purego.go
├── callback_types.go
├── date_time_cgo.go
├── date_time_purego.go
├── engine_cgo.go
├── engine_impl_cgo.go
├── engine_params_cgo.go
├── engine_params_experimental_options.go
├── engine_params_purego.go
├── engine_purego.go
├── error_cgo.go
├── error_codes.go
├── error_go.go
├── error_purego.go
├── executor_cgo.go
├── executor_impl_cgo.go
├── executor_impl_purego.go
├── executor_purego.go
├── http_header_cgo.go
├── http_header_purego.go
├── include_cgo.go
├── internal
│   └── cronet
│       ├── api_purego_float.go
│       ├── api_purego_float_stub.go
│       ├── api_purego.go
│       ├── api_purego_nonwindows.go
│       ├── api_purego_windows.go
│       ├── loader_unix.go
│       ├── loader_windows_float.go
│       ├── loader_windows.go
│       └── symbols_purego.go
├── lib
│   ├── android_386
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── android_amd64
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── android_arm
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── android_arm64
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── darwin_amd64
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── darwin_arm64
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── ios_amd64_simulator
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── ios_arm64
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── ios_arm64_simulator
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── linux_386
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   ├── libcronet.so
│   │   └── LICENSE
│   ├── linux_386_musl
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── linux_amd64
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   ├── libcronet.so
│   │   └── LICENSE
│   ├── linux_amd64_musl
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── linux_arm
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   ├── libcronet.so
│   │   └── LICENSE
│   ├── linux_arm64
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   ├── libcronet.so
│   │   └── LICENSE
│   ├── linux_arm64_musl
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── linux_arm_musl
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── linux_loong64
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   ├── libcronet.so
│   │   └── LICENSE
│   ├── linux_loong64_musl
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── linux_mips64le
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   ├── libcronet.so
│   │   └── LICENSE
│   ├── linux_mipsle
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   ├── libcronet.so
│   │   └── LICENSE
│   ├── linux_mipsle_musl
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── linux_riscv64
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   ├── libcronet.so
│   │   └── LICENSE
│   ├── linux_riscv64_musl
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── tvos_amd64_simulator
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── tvos_arm64
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── tvos_arm64_simulator
│   │   ├── libcronet.a
│   │   ├── libcronet_cgo.go
│   │   ├── libcronet_purego.go
│   │   └── LICENSE
│   ├── windows_amd64
│   │   ├── libcronet.dll
│   │   ├── libcronet.go
│   │   └── LICENSE
│   └── windows_arm64
│       ├── libcronet.dll
│       ├── libcronet.go
│       └── LICENSE
├── LICENSE
├── loader_purego.go
├── Makefile
├── metrics_cgo.go
├── metrics_purego.go
├── naive_client.go
├── naive_conn.go
├── naive_dns.go
├── naive_dns_unix.go
├── naive_dns_windows.go
├── naive_dns_windows_go125.go
├── naive_dns_windows_legacy.go
├── naive_library_cgo.go
├── naive_library_purego.go
├── naive_socket_unix.go
├── naive_socket_windows.go
├── naive_socket_windows_go125.go
├── naive_socket_windows_legacy.go
├── net_error_generated.go
├── net_error.go
├── public_key_pins_cgo.go
├── public_key_pins_purego.go
├── quic_hint_cgo.go
├── quic_hint_purego.go
├── README.md
├── result.go
├── runnable_cgo.go
├── runnable_impl_cgo.go
├── runnable_impl_purego.go
├── runnable_purego.go
├── transport.go
├── types.go
├── upload_data_provider_cgo.go
├── upload_data_provider_handler.go
├── upload_data_provider_impl_cgo.go
├── upload_data_provider_impl_purego.go
├── upload_data_provider_purego.go
├── upload_data_sink_cgo.go
├── upload_data_sink_impl_cgo.go
├── upload_data_sink_purego.go
├── url_request_callback_cgo.go
├── url_request_callback_impl_cgo.go
├── url_request_callback_impl_purego_32bit.go
├── url_request_callback_impl_purego.go
├── url_request_callback_purego.go
├── url_request_cgo.go
├── url_request_finished_info_cgo.go
├── url_request_finished_info_impl_cgo.go
├── url_request_finished_info_listener_cgo.go
├── url_request_finished_info_listener_impl_purego.go
├── url_request_finished_info_listener_purego.go
├── url_request_finished_info_purego.go
├── url_request_impl_cgo.go
├── url_request_params_cgo.go
├── url_request_params_purego.go
├── url_request_purego.go
├── url_request_status_listener_cgo.go
├── url_request_status_listener_impl_cgo.go
├── url_request_status_listener_impl_purego.go
├── url_request_status_listener_purego.go
├── url_response_info_cgo.go
└── url_response_info_purego.go

34 directories, 256 files
```

--- a/.gitignore
+++ b/.gitignore
@@ -8,6 +8,7 @@ CLAUDE.md
 # Prebuilt libraries (committed to go branch only)
 /build-naive
 /lib/
-/include/
+/include/*
+!/include/dummy.go
 /cgo_*.go
 /lib_*_cgo.go

--- a/dummy.go
+++ b/dummy.go
@@ -1,0 +1,12 @@
+//go:build dummy
+
+// This file exists purely to prevent the Go toolchain from stripping
+// away the C source directories and files when `go mod vendor` is used
+// to populate a `vendor/` directory of a project depending on this package.
+
+package cronet
+
+import (
+	// Prevent go tooling from stripping out the c source files.
+	_ "github.com/sagernet/cronet-go/include"
+)

--- a/include/dummy.go
+++ b/include/dummy.go
@@ -1,0 +1,2 @@
+// Package include is a dummy package that prevents go tooling from stripping the C dependencies.
+package include

